spanner.go

```diff
@@ -102,6 +102,11 @@ func DecodeSpannerStructFields(
 		} else if sf.Type == reflect.TypeOf(time.Time{}) && tf.Type == reflect.TypeOf(openapi_types.Date{}) { // Convert from time.Time to openapi_types.Date
 			tfv.Set(reflect.ValueOf(openapi_types.Date{Time: sfv.Interface().(time.Time)}))
 			continue
+		} else if sf.Type == reflect.TypeOf(spanner.NullTime{}) && tf.Type == reflect.TypeOf(&openapi_types.Date{}) { // Convert from spanner.NullTime to *openapi_types.Date
+			if !sfv.Interface().(spanner.NullTime).Valid {
```

Comment:
I think condition is wrong. A helper variable would also be helpful to avoid long casts